File: output_to_file_and_read_all/output_to_file_and_read_all.rb
output_file = "/tmp/output_ruby.txt"
begin
  File.open(output_file, "w") do |wfh|
    wfh.puts("Hello World!\nAdditional line\n")
  end
rescue SystemCallError => e
  puts "class=#{e.class},message=#{e.message}"
rescue IOError => e
  puts "class=#{e.class},message=#{e.message}"
end

File.open(output_file, "r") do |rfh|
  print(rfh.read())
end

File: output_to_file_and_read_all/output_to_file_and_read_all.py
import sys
output_file = "/tmp/output_python.txt"
try:
  with open(output_file, "w") as wfh:
    wfh.write("Hello World!\nAdditional line\n")
except IOError:
  print("Failed to write to " + output_file)
  sys.exit(1)
except:
  print("Unexpected error:", sys.exc_info()[0])
  sys.exit(1)

with open(output_file, "r") as rfh:
  print(rfh.read(), end="")
